test/files/als004/fail-als004-scenario01-wrong_representationidentifier_value.ifc

```diff
@@ -74,6 +73,7 @@ DATA;
 #64=IFCALIGNMENTVERTICAL('0dZodT25PB4PjSqv7D1BCl',#3,'V1',$,$,#380,$);
 #65=IFCRELNESTS('3ZLxvT5W55ARIuY$a2OpaK',#3,'name',$,#64,(#66,#68,#70,#72,#74));
 #66=IFCALIGNMENTSEGMENT('1N_pe3V5P5lBTAZb7Fr6OX',#3,'V1 001 CONSTANTGRADIENT',$,$,#310,#313,#67);
+#34=IFCALIGNMENTSEGMENT('31tSVngQb0XRlPErBWB1XW',#3,'H1 001 LINE',$,$,#179,#182,#35);
```

Comment:
Oh just one more comment. Just moving an instance down in the file doesn't actually change things. At this line the order is defined:

    #33=IFCRELNESTS('0YIHzLddz2kfrHsBRhMEDr',#3,'name',$,#26,(#34,#37,#40,#43,#46,#49,#52,#55,#58,#61));


I do think it's a good idea to test with some files where the error is not on the first instance, but better create a new file and actively modify, e.g:

```
#40=IFCALIGNMENTSEGMENT('2K28_c3_bCK853aPYKlp2m',#3,'H1 003 CIRCULARARC',$,$,#205,#208,#41);
#208=IFCPRODUCTDEFINITIONSHAPE($,$,(#209));
#209=IFCSHAPEREPRESENTATION(#17,'Axis','Segment',(#204));
```

Reply:
Hmm, the order in which context.instances were passed to the handle_then was changed (at index 10 rather than 0). I'm not really understanding why; when I manually check `instances=file.by_type("IfcAlignmentSegment")`, the order is unchanged, as you mentioned.. (#34 still == instances[0])

Changed line #209 into '`Curve2D`' and now the error is at index 2 (as #40 is after #34 and #37; the other instances of type `IfcAlignmentSegment`). Which is both expected and understood. 
[change index fail file for testing](https://github.com/buildingSMART/ifc-gherkin-rules/pull/166/commits/8c6e37241f44844be5dcc2ba24cc0d6334b6ee2b)